### DISCUSSION TITLE
Stärke des Cocktails

### DISCUSSION THREAD

**Discussion by Marvin0795** [AndreWohnsland/CocktailBerry#145] [Ideas]
Anstatt des Gehaltes in %,  wenig, Mittel, stark mit zuvor definierten Werten wählbar machen.

**Comment by AndreWohnsland** [comment]
Würde das generell zu #144 zuordnen, da die beiden Sachen ziemlich ähnlich sind. Mit zuvor definierten Werten meinst du so, wie derzeit, also z.B. 70, 100 und 130 % des Rezeptgehalts an Alkohol oder feste Volumenprozent wie 10, 15, 20 % Alkoholanteil im Getränk? Persönlich würde ich das erste (wegen genereller Umsetzbarkeit und "Charakteristik" des Alkohols bleibt da mehr erhalten)

**Reply by Marvin0795** [reply]
Ja genau, dass passt perfekt zu der #144. Ich würde auch bei den % des Alkohols im Rezept bleiben. 
Einstellung ist Standardmäßig Normal/Mittel, nur wenn man abweichen möchte kann man weniger/mehr klicken.

**Reply by Marvin0795** [reply]
Mal in kurz per Paint gekrizzelt:

![Screenshot 2024-02-12 091703](https://github.com/AndreWohnsland/CocktailBerry/assets/102437306/b23dee5f-c7a3-4d2d-836c-5e8fe139d4f9)

**Reply by AndreWohnsland** [reply]
Ja, so in der Art stelle ich mir das auch vor. Denke bei dem Alkohol ist das relativ einfach umzusetzen, Volumen könnte hier oder da ein bisschen herumzicken :D.

Überlege gerade auch, ob man dann mit mehreren Volumen Optionen in der Config vielleicht das "alte" ganz entfernen kann, und nur noch die Icons hat. Hat für mich den Vorteil, dass ich muss nicht beide Varianten supporten, aber der Nutzer kann trotzdem noch eine gewisse Flexibilität hat, da er seine drei Volumengrößen ja selbst definieren kann. Was aber trotzdem noch da sein sollte, sind die Zahlenanzeigen (Volumen und eigentlich auch Alkoholgehalt), damit der Nutzer weiß, was er bekommt. Wo ich die dann unterbringe, schaue ich dann auch mal.
Was ich leider manchmal habe ist, dass Leute ihr Glasvolumen nicht richtig abschätzen und im Anschluss nochmal ein 100ml Cocktail oder so nach machen. Aber ich denke, sowas könnte man auch mit vorher eingestellten Größen abfangen, muss jeder halt nur einmal auf seine Gläser anpassen. 

Es gibt halt einige andere Teile, die hier noch mitspielen. Es gibt z.B. eine Option, dass der Maker immer das Rezeptvolumen nimmt, und nicht skaliert. Das ändert an der finalen Implementation wenig, heißt aber wie immer für mich mehr Arbeit, damit das alles danach immer noch so funktioniert, wie es vorher getan hat ^^

Die größte Herausforderung ist da meistens am Ende den Platz optimal zu Nutzen, da einige Touchscreens mit 800x480 ziemlich wenig Pixel haben xD.

**Reply by Marvin0795** [reply]
Schön zu sehen wie viel Mühe du dir gibst und es kein Projekt ist, welches nach 1-2 Updates nicht mehr supported wird.. 

Die Definierung der Gläser Größen vorab ist eine gute Sache. 
In unserem Fall hatten wir drei Gläser Größen bereitgestellt 150,250 und 400ml. 
Die meisten haben sich bei einem großen Glas noch 1-2 mal nachgezapft.
In dem Fall wäre es halt super, wenn der Nutzer die drei Größen direkt im Display sieht und wählen kann.
Mit dem Alkoholgehalt konnten die meisten der 50 Leute erst nach der Erklärung etwas anfangen :-D

Die Zahlenanzeigen sehe ich auch als wichtig und könnte in klein vielleicht noch unter die Symbole passen.

Die Größe ist natürlich ein Punkt.. Wir ein 1024*600 7 Zoll eingebaut.. Damit hatten wir noch genug platz..

**Reply by AndreWohnsland** [reply]
Immer doch gerne - ist auch eines meiner Lieblingsprojekte 😄 

Ich schreibe mal n paar Gedanken, vielleicht finden wir zusammen was Gutes:

- Zahlen dazu wären sehr gut, hoffentlich passen die noch unter / neben den Icons rein, ansonsten reicht vielleicht auch der Ansatz beim 2. Punkt
- Ich kann mir auch vorstellen entweder komplett statt dem "Zubereiten / Prepare" die Menge (200 ml etc.) dort anzeigen zu lassen, oder dann halt "200 ml Zubereiten" oder sowas. Hatte auch schon den Fall, dass die Leute sich die falsche Menge gemacht haben, da es oben übersehen wurde. So ist es mehr ersichtlich, was gemacht wird. Die Menge oben wird gerne auch mal übersehen ...
- Alkoholgehalt seh ich auch so, häufig wurde der ignoriert. Vielleicht braucht es keine Zahl, zwei optionale Buttons "Stark / Schwach" könnten dann schon reichen, wenn man mit der "Standardausführung" des Cocktails unzufrieden ist.
- Ich selbst, sowie viele anderen Nutzer haben ein 1024x600 7-Zoller. Leider ist das offizielle Rapsberry Pi 7-Zoll Touch mit 800x480, ansonsten würde ich direkt einfach sagen, ich supporte nur ab 1000, aber da es halt das "Standard Touch" ist, machts n wenig schwer. 

Je mehr ich im Allgemeinen darüber nachdenke, desto mehr will ich an sich auch statt den plus / minus / Zahl einfache Buttons haben, der Teufel liegt am Ende dann immer im Detail, wie UI/UX am besten ist und natürlich wie ich das am besten integrierbar und auch Zukunftssicher / kompatibel mache.

**Reply by Marvin0795** [reply]
Punkt 1 -> Ja das könnte schwierig werden..
Punkt 2 -> Genau diesen Fall hatten wir auch, auf die Zahlen oben hat kaum einer geachtet.. Dein Ansatz hört sich gut an, 
                  anstatt des Prepare Buttons die einzelnen größen anzuwählen. Viele möchten einen Cocktail auch erst probieren, da ist z.B. 
                  eine kleine größe wichtig.
                  Ich würde dennoch Icons einfügen, es muss ja nur eine größen Andeutung sein, da viele nicht genau wissen, in 
                  welches Glas wie viel passt. 
Punkt 3 -> Sehe ich auch so!
Punkt 4 -> Aufgrund der Standard Größe würde ich das auch so machen. Ein bisschen Variabel muss es schon noch sein..

**Reply by AndreWohnsland** [reply]
Bin gerade an der ersten Implementation. Man kann jetzt eine Liste an Volumen in der Konfiguration einstellen, die vorgeschlagen werden:
![tmp_ui](https://github.com/AndreWohnsland/CocktailBerry/assets/50302161/06e8bddb-b81e-4a7b-a982-c1a0a7a48c2f)

Bin hier am Überlegen, die Einheit wegzulassen. Das wäre gut, wenn man vier oder mehr Volumen verwenden will.

Wenn nur ein Element, oder Rezeptvolumen eingestellt ist, sieht man den alten Button / Layout:
![tmp_ui2](https://github.com/AndreWohnsland/CocktailBerry/assets/50302161/44e29546-b788-42e4-8ab6-8dc16e0347cb)

**Reply by Marvin0795** [reply]
Das sieht doch schon super aus! Hab ich nichts dran auszusetzen :-D
